Lib/test/test_pstats.py

```diff
@@ -30,12 +34,39 @@ class StatsTestCase(unittest.TestCase):
     def setUp(self):
         stats_file = support.findfile('pstats.pck')
         self.stats = pstats.Stats(stats_file)
+        to_compile = 'import os'
+        self.temp_storage = tempfile.NamedTemporaryFile(delete=False)
+        profiled = compile(to_compile, '<string>', 'exec')
+        cProfile.run(profiled, filename=self.temp_storage.name)
```

Comment:
You don't need to compile the string, `cProfile` takes a string.

Reply:
sure.